pkgs/development/libraries/vte/default.nix

```diff
@@ -54,8 +55,11 @@ stdenv.mkDerivation rec {
     pcre2
     zlib
     icu
-    systemd
-  ];
+  ] ++ lib.optionals stdenv.isLinux [ systemd ];
+
+  mesonFlags = lib.optionals stdenv.cc.isClang [ "-D_b_symbolic_functions=false" ];
+
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-I${libcxx}/include/c++/v1";
```

Comment:
What error does this fix?

Reply:
It doesn't find `cstdio` header:
```
...
[27/100] Compiling C++ object 'src/25a6634@@vte-urlencode-cwd@exe/urlencode.cc.o'ypebuiltins.cc.o'
FAILED: src/25a6634@@vte-urlencode-cwd@exe/urlencode.cc.o
clang++ -Isrc/25a6634@@vte-urlencode-cwd@exe -Isrc -I../src -I. -I.. -Xclang -fcolor-diagnostics -pipe -std=gnu++17 -Wall -Wextra -Wcast-align -Wempty-body -Wendif-labels -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Wfloat-equal -Wignored-qualifiers -Winvalid-pch -Wmissing-declarations -Wmissing-field-initializers -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wno-address-of-packed-member -Wno-missing-field-initializers -Wno-packed -Wno-switch-enum -Wno-unused-parameter -Wshadow -Wshift-negative-value -Wsign-compare -Wstrict-aliasing=2 -Wtype-limits -Wundef -Wuninitialized -Wuninitialized -Wunused -Wunused-function -Wunused-label -Wunused-local-typedefs -Wunused-value -Wunused-variable -Wvla -Wwrite-strings -fdiagnostics-show-option -fno-common -fstack-protector -fstack-protector-strong -fno-strict-aliasing -Wnon-virtual-dtor -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -fvisibility=hidden -Werror=format=2 -Werror=format-nonliteral -Werror=format-security -nostdlib -MD -MQ 'src/25a6634@@vte-urlencode-cwd@exe/urlencode.cc.o' -MF 'src/25a6634@@vte-urlencode-cwd@exe/urlencode.cc.o.d' -o 'src/25a6634@@vte-urlencode-cwd@exe/urlencode.cc.o' -c ../src/urlencode.cc
../src/urlencode.cc:22:10: fatal error: 'cstdio' file not found
#include <cstdio>
         ^~~~~~~~
1 error generated.
[28/100] Compiling C++ object 'src/25a6634@@vte-2.91@sha/pty.cc.o'
In file included from ../src/pty.cc:76:
../src/glib-glue.hh:160:21: warning: private field 'm_name' is not used [-Wunused-private-field]
        char const* m_name{nullptr};
                    ^
1 warning generated.
[33/100] Compiling C++ object 'src/25a6634@@test-parser@exe/parser.cc.o'
...
```
I saw it being used in various expressions in nixpkgs, but I'm not sure whether is the right way to solve it